src/implementations/react/src/elements/components/FormElements/Dropdown.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import DropdownAdapter
+  from '../../../adapters/FormElements/DropdownAdapter.js';
+import * as PropTypes from 'prop-types';
+
+const OptionAdapter = DropdownAdapter.Option;
+
+class Dropdown extends Component {
+
+  static propTypes = {
+    label: PropTypes.string,
+    instructions: PropTypes.string,
+    placeholder: PropTypes.string,
+    disabled: PropTypes.bool,
+    required: PropTypes.string,
+    selectedOption: PropTypes.string,
+
+    options: PropTypes.arrayOf(PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.string
+    })),
+
+    onChange: PropTypes.func,
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    onKeypress: PropTypes.func
+  }
```

Comment:
Form elements in React need to support two modes—"controlled" and "uncontrolled". Controlled means that if a `value` prop is passed in (or in this case,  `selectedOption`) the `value` MUST be fixed to the passed in value. Uncontrolled means the developer optionally sets the initial state, and user interaction modifies the state from there. This PR supports the uncontrolled mode, but needs controlled also.

The convention is to have a `value` prop that always takes precedence over state if passed (for controlled use), and an `initialValue` prop that sets the initial state (for uncontrolled use).

Heavy-duty apps are almost certainly going to use it in the controlled way, like this (With redux it would look different, but would use the same props on Dropdown):
```
class MyComponent extends Component {
  constructor (props) {
    super(props);
    this.state = {
      selectedOption: 0
    };
  }

  setSelectedOption(selectedOptionId) {
    this.setState({ selectedOption: selectedOptionId });
    // Do side effect of selection change
  }

  render() {
    return (
      <Dropdown
        options={this.props.options}
        selectedOption={this.state.selectedOption}
        onChange={this.setSelectedOption}
      />
    );
  }
}
```

I suggest adding `initialSelectedOption` to set initial state and making `selectedOption` override state.

Check out `getDefaultValue` and `getRenderedValue` in TextField.js for examples.